src/guide/agent-functions/index.rst

```diff
@@ -7,8 +7,12 @@ As FLAME GPU is an agent-based modelling framework, agent functions are central
 
 Each agent has one or more agent functions, which allow agents to interact with the environment and other agents. Additionally, agent functions can be conditionally executed based on agent states and properties.
 
-Each agent function in FLAME GPU 2 is associated with a particular agent type and is represented by an :class:`AgentFunctionDescription<flamegpu::AgentFunctionDescription>` object. This object describes the name of the function, the agent state in which it should be active, and any state transitions it should apply to the agent. The implementation code for the behaviour is written separately and is identified by using the
-:c:macro:`FLAMEGPU_AGENT_FUNCTION` macro to label the code. Alternatively, the implementations can be written in strings and loaded at runtime. The latter method is the only method supported by FLAME GPU 2's Python API.
+Each agent function in FLAME GPU 2 is associated with a particular agent type and is represented by an :class:`AgentFunctionDescription<flamegpu::AgentFunctionDescription>` object. This object describes the name of the function, the agent state in which it should be active, and any state transitions it should apply to the agent. 
+
+The implementation code for the behaviour (the actual agent function) can be specified in three alternative ways;
```

Comment:
List does not render correctly due to missing preceding new line

![image](https://user-images.githubusercontent.com/628937/186685731-22a25f6c-64da-44fd-923c-ab682bf1374e.png)
```suggestion
The implementation code for the behaviour (the actual agent function) can be specified in three alternative ways;

```